builder/azure/common/template/template_builder.go

```diff
@@ -517,6 +516,15 @@ func (s *TemplateBuilder) SetSecurityProfile(secureBootEnabled bool, vtpmEnabled
 	return nil
 }
 
+func (s *TemplateBuilder) SetSpecializedVM() error {
```

Comment:
So my question around the child being specialized if the  parent is specialized stems from this function. It's name indicates that it is setting the vm as specialized but it is just removing the OsProfile settings. If this function will solely just clear the OsProfile what do you think about naming it differently. Maybe something like 
```suggestion
func (s *TemplateBuilder) ClearOsProfile() error {
```

Reply:
Sounds good to me, I'll make that change!